pkg/apiserver/registry/networkpolicy/clustergroupmember/rest_test.go

```diff
@@ -162,3 +213,163 @@ func TestRESTGet(t *testing.T) {
+				CurrentPage:  1,
+			},
+			expectedErr: false,
+		},
+		{
+			name:              "page2/2-group-member-pagination",
+			groupName:         "cgA",
+			paginationOptions: &controlplane.PaginationGetOptions{Page: 2, Limit: 2},
+			expectedObj: &controlplane.ClusterGroupMembers{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cgA",
+				},
+				EffectiveMembers: []controlplane.GroupMember{
+					{
+						Pod: &controlplane.PodReference{
+							Name:      "pod3",
+							Namespace: "ns1",
+						},
+						IPs: []controlplane.IPAddress{
+							[]byte{127, 10, 0, 1},
+						},
+					},
+				},
+				TotalMembers: 3,
+				TotalPages:   2,
+				CurrentPage:  2,
+			},
+			expectedErr: false,
+		},
+		{
+			name:              "exceed-page-group-member-pagination",
+			groupName:         "cgA",
+			paginationOptions: &controlplane.PaginationGetOptions{Page: 5, Limit: 2},
+			expectedObj: &controlplane.ClusterGroupMembers{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cgA",
+				},
+				EffectiveMembers: []controlplane.GroupMember{},
+				TotalMembers:     3,
+				TotalPages:       2,
+				CurrentPage:      5,
+			},
+			expectedErr: false,
+		},
+		{
+			name:              "default-zero-group-member-pagination",
+			groupName:         "cgA",
+			paginationOptions: &controlplane.PaginationGetOptions{Page: 0, Limit: 0},
+			expectedObj: &controlplane.ClusterGroupMembers{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cgA",
+				},
+				EffectiveMembers: []controlplane.GroupMember{
+					{
+						Pod: &controlplane.PodReference{
+							Name:      "pod3",
+							Namespace: "ns1",
+						},
+						IPs: []controlplane.IPAddress{
+							[]byte{127, 10, 0, 1},
+						},
+					},
+					{
+						Pod: &controlplane.PodReference{
+							Name:      "pod2",
+							Namespace: "ns1",
+						},
+						IPs: []controlplane.IPAddress{
+							[]byte{127, 10, 0, 1},
+						},
+					},
+					{
+						Pod: &controlplane.PodReference{
+							Name:      "pod1",
+							Namespace: "ns1",
+						},
+						IPs: []controlplane.IPAddress{
+							[]byte{127, 10, 0, 1},
+						},
+					},
+				},
+				TotalMembers: 3,
+				TotalPages:   1,
+				CurrentPage:  1,
+			},
+			expectedErr: false,
+		},
+		{
+			name:              "err-page-group-member-pagination",
+			groupName:         "cgA",
+			paginationOptions: &controlplane.PaginationGetOptions{Page: -1, Limit: 2},
+			expectedErr:       true,
+		},
+		{
+			name:              "err-limit-group-member-pagination",
+			groupName:         "cgA",
+			paginationOptions: &controlplane.PaginationGetOptions{Page: 1, Limit: -2},
+			expectedErr:       true,
+		},
+	}
+	rest := NewREST(fakeQuerier{members: getTestMembersPagination()})
+	for _, tt := range tests {
+		actualGroupList, err := rest.Get(request.NewDefaultContext(), tt.groupName, tt.paginationOptions)
+		if tt.expectedErr {
+			require.Error(t, err)
+		} else {
+			require.NoError(t, err)
+		}
+		if tt.name == "default-zero-group-member-pagination" {
```

Comment:
Because in default zero case, no pagination is performed thus the three returned members are sorted. `assert.ElementsMatch` is needed separately to resolve flakiness.

Reply:
Maybe we should sort in the non-pagination case as well for consistency. I don't think the performance argument in that case is a very valid one.